File: skills/smoke-alarm/tests/fixtures/ts/examples.test.ts
// Labeled fixtures for the smoke-alarm classifier self-test.
// Each test title encodes its expected category: ..._expect_<CAT>.

function add(a: number, b: number): number {
  return a + b;
}

describe('add', () => {
  it('value equality expect_S1', () => {
    expect(add(2, 3)).toBe(5);
  });

  it('error check expect_S2', () => {
    expect(() => {
      throw new Error('boom');
    }).toThrow();
  });

  it('strong combo expect_S3', () => {
    expect(add(2, 2)).toBe(4);
    expect(() => JSON.parse('{')).toThrow();
  });

  it('no assertion expect_W1', () => {
    const result = add(1, 1);
    console.log(result);
  });

  it('defined only expect_W2', () => {
    expect(add(1, 1)).toBeDefined();
  });

  it('mock only expect_W4', () => {
    const spy = jest.fn();
    spy();
    expect(spy).toHaveBeenCalled();
  });

  it('snapshot only expect_W5', () => {
    expect(add(1, 2)).toMatchSnapshot();
  });
});

File: skills/smoke-alarm/tests/fixtures/go/examples_test.go
// Labeled fixtures for the smoke-alarm classifier self-test.
// Each test name encodes its expected category: Test..._Expect_<CAT>.
package examples

import (
	"errors"
	"testing"
)

func add(a, b int) int { return a + b }

func TestValueEquality_Expect_S1(t *testing.T) {
	if got := add(2, 3); got != 5 {
		t.Fatalf("got %d", got)
	}
}

func TestErrorCheck_Expect_S2(t *testing.T) {
	err := errors.New("boom")
	if !errors.Is(err, err) {
		t.Fatal("expected match")
	}
}

func TestNoAssertion_Expect_W1(t *testing.T) {
	_ = add(1, 1)
}

func TestNotNil_Expect_W2(t *testing.T) {
	var p *int
	if p == nil {
		_ = p
	}
}
